MDANSE/Src/MDANSE/Framework/Jobs/TrajectoryFilter.py

```diff
@@ -0,0 +1,256 @@
+                    "time_step_ps": self.configuration["trajectory"]["md_time_step"],
+                }
+            )
+
+        filter = filter_class(**filter_attributes)
+
+        trajectories = copy.deepcopy(self.atomic_trajectory_array)
+
+        # Apply filter (only apply initial position offset to atoms if filter is not lowpass and setting has been applied)
+        filtered_coords = apply(
+            filter,
+            trajectories,
+            apply_offsets=filter_attributes.get("attenuation_type", "bandpass")
+            is not "lowpass",
+        )
+
+        # Create trajectory writer object
+        self._output_trajectory = TrajectoryWriter(
+            self.configuration["output_files"]["file"],
+            self.configuration["trajectory"]["instance"].chemical_system,
+            filter_attributes["n_steps"],
+            None,
+            positions_dtype=self.configuration["output_files"]["dtype"],
+            compression=self.configuration["output_files"]["compression"],
+        )
+
+        # Write trajectory
+        write_filtered_trajectory(
+            parent_configuration=self.configuration,
+            nsteps=filter_attributes["n_steps"],
+            filtered_coordinates=filtered_coords,
+            output_trajectory=self._output_trajectory,
+        )
+
+        # The input trajectory is closed.
+        self.configuration["trajectory"]["instance"].close()
+
+        # The output trajectory is closed.
+        self._output_trajectory.close()
+
+        # Write the filter metadata to output
+        outputFile = h5py.File(self.configuration["output_files"]["file"], "r+")
+        outputFile.create_group("metadata").create_dataset(
+            "trajectory_filter",
+            (1,),
+            data=filter.__str__(),
+            dtype=h5py.string_dtype(),
+        )
+
+        outputFile.close()
+
+        super().finalize()
+
+
+def apply(filter, trajectories, apply_offsets: bool) -> np.ndarray:
+    """ """
+    output_trajectory_array = np.zeros(trajectories.shape)
+
+    for at, atom in enumerate(trajectories):
+        x, y, z = atom
+
+        # Store initial positions
+        offsets = np.array([x[0], y[0], z[0]])
+
+        for i, component in ((0, x), (1, y), (2, z)):
+            output = filter.apply(component)
+
+            if apply_offsets:
+                output += offsets[i]
+            output_trajectory_array[at][i] = output
+
+    return output_trajectory_array
+
+
+def write_filtered_trajectory(
+    parent_configuration,
+    nsteps: int,
+    filtered_coordinates: np.ndarray,
+    output_trajectory: TrajectoryWriter,
+) -> None:
+    """ """
+    time = parent_configuration["frames"]["time"]
+    dt = time[1] - time[0]
+    for index in range(nsteps):
+        frame_coordinates = list()
+        for atom in filtered_coordinates:
+            frame_coordinates.append((atom[0][index], atom[1][index], atom[2][index]))
```

Comment:
```suggestion
        frame_coordinates = [(atom[0][index], atom[1][index], atom[2][index])
                             for atom in filtered_coordinates]
```
Alternatively if there are only 3 elements:
```suggestion
        frame_coordinates = [(a[index], b[index], c[index])
                             for (a, b, c) in filtered_coordinates]
```